CMakeLists.txt

```diff
@@ -15,6 +15,8 @@ endfunction()
 
 if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
     add_compile_options(-Wall -Wextra -pedantic -Wcast-qual -Wfloat-equal -Wmissing-declarations -Wmissing-format-attribute -Wredundant-decls -Wshadow -Wundef -Wold-style-cast -Wno-multichar)
+elseif (CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+    add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
```

Comment:
These warnings should be "fixed" by specifying `_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1` instead - see https://learn.microsoft.com/en-us/cpp/c-runtime-library/secure-template-overloads.

I wasn't aware of this so it might have been introduced only in recent years.

Reply:
I'm not so sure about this, it seems to be the usual MSVC scaring people into using _s functions when the 'normal' C standard functions can be used just fine if used correctly. _CRT_SECURE_NO_WARNINGS is just the broader macro that tells MSVC to stop complaining.